src/connector/src/source/cdc/external/postgres.rs

```diff
@@ -256,6 +257,56 @@ impl PostgresExternalTableReader {
     }
 }
 
+pub fn type_name_to_pg_type(ty_name: &str) -> Option<PgType> {
+    match ty_name.to_lowercase().as_str() {
+        "smallint" | "int2" => Some(PgType::INT2),
+        "integer" | "int" | "int4" => Some(PgType::INT4),
+        "bigint" | "int8" => Some(PgType::INT8),
+        "real" | "float4" => Some(PgType::FLOAT4),
+        "double precision" | "float8" => Some(PgType::FLOAT8),
+        "numeric" | "decimal" => Some(PgType::NUMERIC),
+        "boolean" | "bool" => Some(PgType::BOOL),
+        "varchar" | "character varying" => Some(PgType::VARCHAR),
+        "char" | "character" | "bpchar" => Some(PgType::BPCHAR),
```

Comment:
This is the ambiguous part and why we should know whether upstream is sending `smallint` or `int2` - it cannot be both.
|oid|display|typname|
|-|-|-|
|21|smallint|int2|
|1042|char|bpchar|
|18|"char"|char|

Without knowing whether we are parsing display name or internal typname, we do not know whether `char` means 1042 or 18, although both would eventually be mapped to RisingWave's varchar.